drivers/pinctrl/pinctrl_bflb_bl60x_70x.c

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright (c) 2025 MASSDRIVER EI (massdriver.space)
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/drivers/pinctrl.h>
+
+#include <bflb_soc.h>
+#include <glb_reg.h>
+#include <zephyr/dt-bindings/pinctrl/bflb-common-pinctrl.h>
+
+#if defined(CONFIG_SOC_SERIES_BL60X)
+#include <zephyr/dt-bindings/pinctrl/bl60x-pinctrl.h>
+#else
+#error "Unsupported Platform"
+#endif
+
+void pinctrl_bflb_configure_uart(uint8_t pin, uint8_t func)
+{
+	/* uart func for BL602 and BL702 Only*/
+	uint32_t regval;
+	uint8_t sig;
+	uint8_t sig_pos;
+
+	regval = sys_read32(GLB_BASE + GLB_UART_SIG_SEL_0_OFFSET);
+
+	sig = pin % 8;
+	sig_pos = sig << 2;
+
+	regval &= (~(0x0f << sig_pos));
+	regval |= (func << sig_pos);
+
+	for (uint8_t i = 0; i < 8; i++) {
+		/* reset other sigs which are the same with uart_func */
+		sig_pos = i << 2;
+		if (((regval & (0x0f << sig_pos)) == (func << sig_pos)) && (i != sig) && (func !=
+0x0f)) {
+			regval &= (~(0x0f << sig_pos));
+			regval |= (0x0f << sig_pos);
+		}
+	}
+
+	sys_write32(regval, GLB_BASE + GLB_UART_SIG_SEL_0_OFFSET);
+}
+
+void pinctrl_bflb_init_pin(pinctrl_soc_pin_t pin)
+{
+	uint8_t drive;
+	uint8_t function;
+	uint16_t mode;
+	uint32_t regval;
+	uint8_t real_pin;
+	uint8_t is_odd = 0;
+	uint32_t cfg = 0;
+	uint32_t cfg_address;
+
+	real_pin = BFLB_PINMUX_GET_PIN(pin);
+	function = BFLB_PINMUX_GET_FUN(pin);
+	mode = BFLB_PINMUX_GET_MODE(pin);
+	drive = BFLB_PINMUX_GET_DRIVER_STRENGTH(pin);
+
+	/* Disable output anyway */
+	regval = sys_read32(GLB_BASE + GLB_GPIO_CFGCTL34_OFFSET + ((real_pin >> 5) << 2));
+	regval &= ~(1 << (pin & 0x1f));
```

Comment:
real_pin?

Reply:
uh, yea? Would you prefer something like 'pin_number' ?

Reply:
pin is already used as the input to the function, this is the pin number extracted from that

Reply:
Ah no wait sorry, let me check. 1f should be function